pages/setup/account.page.tsx

```diff
@@ -0,0 +1,146 @@
+// the user has more than one account list and don't have a default chosen yet.
+const AccountPage: React.FC<PageProps> = ({ accountListOptions }) => {
+  const { t } = useTranslation();
+  const { appName } = useGetAppSettings();
+  const { push } = useRouter();
+  const [updateUserDefaultAccount, { loading: isSubmitting }] =
+    useUpdateUserDefaultAccountMutation();
+
+  const [defaultAccountList, setDefaultAccountList] =
+    useState<AccountList | null>(null);
+
+  const handleSave = async () => {
+    if (!defaultAccountList) {
+      return;
+    }
+
+    await updateUserDefaultAccount({
+      variables: {
+        input: {
+          attributes: {
+            defaultAccountList: defaultAccountList.id,
+          },
+        },
+      },
+    });
+    push(`/accountLists/${defaultAccountList.id}/settings/preferences`);
+  };
+
+  return (
+    <>
+      <Head>
+        <title>
+          {appName} | {t('Setup - Default Account')}
+        </title>
+      </Head>
+      <SetupPage title={t('Set default account')}>
+        {t(
+          'Which account would you like to see by default when you open {{appName}}?',
+          { appName },
+        )}
+        <Autocomplete
+          autoHighlight
+          value={defaultAccountList}
+          onChange={(_, value) => setDefaultAccountList(value)}
+          options={accountListOptions.accountLists.nodes}
+          getOptionLabel={(accountList) => accountList.name ?? ''}
+          fullWidth
+          renderInput={(params) => (
+            <TextField {...params} label={t('Account')} />
+          )}
+        />
+        <LargeButton
+          variant="contained"
+          fullWidth
+          onClick={handleSave}
+          disabled={!defaultAccountList || isSubmitting}
+        >
+          {t('Continue Tour')}
+        </LargeButton>
+      </SetupPage>
+    </>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps<PageProps> = async (
+  context,
+) => {
+  const session = await getSession(context);
+  const apiToken = session?.user?.apiToken;
+  if (!apiToken) {
+    return {
+      redirect: {
+        destination: '/login',
+        permanent: false,
+      },
+    };
+  }
+
+  const ssrClient = makeSsrClient(apiToken);
+  const { data: accountListOptions } =
+    await ssrClient.query<AccountListOptionsQuery>({
+      query: AccountListOptionsDocument,
+    });
+
+  if (accountListOptions.accountLists.nodes.length === 1) {
+    // The user has exactly one account list, so set it as the default and go to preferences
+    const defaultAccountListId = accountListOptions.accountLists.nodes[0].id;
+    await ssrClient.mutate<
+      UpdateUserDefaultAccountMutation,
+      UpdateUserDefaultAccountMutationVariables
+    >({
+      mutation: UpdateUserDefaultAccountDocument,
+      variables: {
+        input: {
+          attributes: {
+            defaultAccountList: defaultAccountListId,
+          },
+        },
+      },
+    });
```

Comment:
just a though, This might not be needed but if this request errored, it would cause the page would error out, I wonder if it were to error, to show the page and let the user choose. 

Feel free to disagree.

Reply:
Good point, I should definitely do that. They'll presumably get the same error when they try to pick one on the page, but at least they'll get a helpful message.